pkg/controller/util.go

```diff
@@ -296,3 +300,23 @@ func podPhaseFromPVC(pvc *v1.PersistentVolumeClaim) v1.PodPhase {
 func podSucceededFromPVC(pvc *v1.PersistentVolumeClaim) bool {
 	return (podPhaseFromPVC(pvc) == v1.PodSucceeded)
 }
+
+func setConditionFromPod(anno map[string]string, pod *v1.Pod) {
```

Comment:
This cannot be used for upload?

Reply:
Its because I moved
```
anno[AnnPodRestarts] = strconv.Itoa(int(pod.Status.ContainerStatuses[0].RestartCount))
```
In here, which is different for upload. But I think that is a mistake, if I leave it out I can re-use the rest in upload